capture/crawler_estadao.py

```diff
@@ -0,0 +1,247 @@
+
+# add formatter to ch
+ch.setFormatter(formatter)
+fh.setFormatter(formatter)
+
+# add ch to logger
+logger.addHandler(ch)  # uncomment for console output of messages
+logger.addHandler(fh)
+
+client = pymongo.MongoClient(settings.MONGOHOST, 27017)
+MCDB = client.MCDB
+ARTICLES = MCDB.articles  # Article Collection
+ARTICLES.ensure_index("source")
+
+
+def find_articles(category, page=1):
+	"""Get the urls of last news and its categories
+
+	:param category: the category of the news (politica, economia, internacional, esportes, sao-paulo, cultura)
+	:param page: the page number of last news
+	:type category: string
+	:type page: integer
+	:return: last news with its categories
+	:rtype: list()
+
+	"""
+
+	CATEGORIES = (u'politica', u'economia', u'internacional', u'esportes', u'sao-paulo', u'cultura', u'opiniao',
+				  u'alias', u'brasil', u'ciencia', u'educacao', u'saude', u'sustentabilidade', u'viagem')
+
+	if category not in CATEGORIES:
+		raise ValueError("Category value not accepted.")
+
+	params = (category, page)
+
+	INDEX_URL = "http://{0}.estadao.com.br/ultimas/{1}".format(*params)
+
+	index = requests.get(INDEX_URL).content
+	soup = BeautifulSoup(index)
+	news_index = soup.findAll("div", {"class":"listadesc"})
+	news_urls = [url.contents[1]['href'] for url in news_index]
+	
+	return news_urls
+
+def clean_content(response_content):
+	""" For getting only the news' content is necessary to remove dirty elements like script,
+		iframe, figcaption tags and related news reference in content's body.
+	"""
+	
+	soup = BeautifulSoup(response_content)
+				
+	for tag in soup.find_all("header"):
+		tag.decompose()
+	for tag in soup.find_all("div", {"class":"wp-caption"}):
+		tag.decompose()
+	for tag in soup.find_all("div", {"class":"tags"}):
+		tag.decompose()
+	for tag in soup.find_all('code'):
+		tag.decompose()
+	for tag in soup.find_all('iframe'):
+		tag.decompose()
+	for tag in soup.find_all('div', {'class':'relacionadastexto'}):
+		tag.decompose()
+	for tag in soup.find_all('figcaption'):
+		tag.decompose()
+	for tag in soup.find_all('script'):
+		tag.decompose()
+	return soup.html
+
+def extract_published_time(url, soup):
+	""" Get the news published datetime 	
+
+	:param soup: object with news html page
+	:type soup: BeautifulSoup object
+	:return: news published datetime
+	:rtype: string
+
+	"""
+
+	MONTHS = {	u"janeiro": u"Jan", u"fevereiro": u"Fev", u"mar\xe7o": u"Mar", u"abril": u"Apr",
+				u"maio": u"May", u"junho": u"Jun", u"julho": u"Jul", u"agosto": u"Aug",
+				u"setembro": u"Sep", u"outubro": u"Oct", u"novembro": u"Nov", u"dezembro": u"Dec"
+			 }
+
+	try:
+		time = soup.findAll("p", {"class":"data"})[0].text
+	except:
+		try:
+			time = soup.findAll("span", {"class":"data"})[0].text
+		except:
+			logger.error('wrong data tags')
+			return None
+	
+	try:
+		time = time.strip().split()
+		time[1] = time[1].lower()
+		time[1] = MONTHS[time[1]]
+		
+		if url.find("estadao.com.br/noticias/") is not -1:
```

Comment:
I think `if "estadao.com.br/noticias/" in url:` is a little more idiomatic here.